apps/andi/lib/andi/input_schemas/ingestion.ex

```diff
@@ -51,71 +51,137 @@ defmodule Andi.InputSchemas.Ingestion do
     :name
   ]
 
+  def validate(%Ecto.Changeset{data: %__MODULE__{}} = changeset) do
+    data_as_changes =
```

Comment:
Validate does a few important things:
1 - Get the underlying data + any pending changes into a map as changes

2 - Recasts the existing data on top of itself with force_changes: true
2.1 - force_changes: true is extremely important. We needed to upgrade ecto to get this flag.
2.2 - Apply changes over an empty changeset is invalid. It would indicate that all children are new and try to insert them, since essentially you're telling the database I'm changing from nothing to something.
2.3 - Only changes are validated, not underlying data. So validating draft data is impossible since it has already been saved to the database
2.4 - force_changes: true is a framework-supported workaround that allows us to force draft changes as "changes", which will update everything in place rather than try to recreate it.

3 - Uses children changeset function to cast_assoc, which includes validations. Once we refactor the children, we will need to use the validate function instead, or maybe a changeset_with_validation if that makes more sense